src/lib.rs

```diff
@@ -185,6 +195,42 @@ impl<T: Runtime> Clone for Client<T> {
             metadata: self.metadata.clone(),
             runtime_version: self.runtime_version.clone(),
             _marker: PhantomData,
+            page_size: self.page_size,
+        }
+    }
+}
+
+/// Iterates over key value pairs in a map.
+pub struct KeyIter<T: Runtime, F: Store<T>> {
+    client: Client<T>,
+    _marker: PhantomData<F>,
+    count: u32,
+    hash: Option<T::Hash>,
+    start_key: Option<StorageKey>,
+    keys: Vec<StorageKey>,
+}
+
+impl<T: Runtime, F: Store<T>> KeyIter<T, F> {
+    /// Returns the next key value pair from a map.
+    pub async fn next(&mut self) -> Result<Option<(StorageKey, F::Returns)>, Error> {
```

Comment:
I'm wondering whether it's worth making this `impl Stream` too? Then you could do e.g. `client.account_iter().collect::<Vec<_>>().await`

Reply:
Would be nice, but it seems terribly annoying to do. Maybe I'm missing a simple solution, but basically once you go from manually implemented futures and streams to async syntax there isn't an easy way back.

Reply:
Yeah. I can't think of a nice and elegant way to do it - maybe a future PR